ED/protocol.html

```diff
@@ -1067,7 +1067,7 @@ <h4 property="schema:name">Modifying Resources Using N3 Patches</h4>
                       <li id="server-patch-n3-single" rel="spec:requirement" resource="#server-patch-n3-single"><span property="spec:statement">The patch document <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> contain exactly one patch resource, identified by one or more of the triple patterns described above, which all share the same <code>?patch</code> subject.</span></li>
                       <li id="server-patch-n3-simple-type" rel="spec:requirement" resource="#server-patch-n3-type">A patch resource <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> contain a triple <code><span property="spec:statement">?patch rdf:type solid:InsertDeletePatch</span></code>.</li>
                       <li id="server-patch-n3-variables" rel="spec:requirement" resource="#server-patch-n3-variables"><span property="spec:statement">The <code>?insertions</code> and <code>?deletions</code> formulae <span rel="spec:requirementLevel" resource="spec:MUSTNOT">MUST NOT</span> contain variables that do not occur in the <code>?conditions</code> formula.</span></li>
-                      <li id="server-patch-n3-blank-nodes" rel="spec:requirement" resource="#server-patch-n3-blank-nodes"><span property="spec:statement">The <code>?insertions</code> and <code>?deletions</code> formulae <span rel="spec:requirementLevel" resource="spec:MUSTNOT">MUST NOT</span> contain blank nodes.</span></li>
+                      <li id="server-patch-n3-blank-nodes" rel="spec:requirement" resource="#server-patch-n3-blank-nodes"><span property="spec:statement">The <code>?deletions</code> formulae <span rel="spec:requirementLevel" resource="spec:MUSTNOT">MUST NOT</span> contain blank nodes.</span></li>
```

Comment:
```suggestion
                      <li id="server-patch-n3-blank-nodes" rel="spec:requirement" resource="#server-patch-n3-blank-nodes"><span property="spec:statement">The <code>?deletions</code> formula <span rel="spec:requirementLevel" resource="spec:MUSTNOT">MUST NOT</span> contain blank nodes.</span></li>
```

Reply:
Is this actually singular `formula` (being only `?deletions`, and not `?insertions and ?deletions`) or might there be multiple `?deletions formulae` (which is how I read it)?

Reply:
Ah right, in that case would still be plural, yes. But notice how, one line above it, `<code>?conditions</code> formula` is singular and see https://github.com/solid/specification/pull/711/files/bf731cbd8afb19cf216224a9972e37715edc0fd8#diff-d28e1706bf82e98cbda64de615436823ef8b0b61f5c05b2327b35bb1d0863558R1057-R1061 for context of how the `?inserts`, `?deletes` and `?conditions` formulae are introduced, I do feel it should be changed to singular 'formula' here.

Reply:
But, also in line 1069, `?insertions and deletions formulae` is plural, and I think that's the one that parallels the `formula/e` we're discussing in line 1070, as these are both part of the subject of these sentences.

The one you highlight in line 1069 is part of the object of that sentence. Looking carefully at *that* sentence, I think that the concluding `formula` *should* actually be written as "optionally singular or plural", i.e., `formula(e)` or `formula/e`.

Reply:
Committing @michielbdejong's `formula` suggestion as this aligns with the reference to conditions `formula` in the above line as the optionally singular formula(e) doesn't appear to be elsewhere in the document.

I suggest proposing a set of batch changes if there is a strong opinion that this should be updated